src/grids.jl

```diff
@@ -93,19 +238,72 @@ one for each face of the octahedron. E.g. 20,24,28,32,...nlon-4,nlon,nlon,nlon-4
 The maximum number of longitue points is `nlon`. The values of all grid points are stored in a vector
 field `v` that unravels the data 0 to 360˚, then ring by ring, which are sorted north to south."""
 struct OctahedralGaussianGrid{T} <: AbstractOctahedralGrid{T}
-    v::Vector{T}    # data vector, ring by ring, north to south
+    data::Vector{T}    # data vector, ring by ring, north to south
     nlat_half::Int  # number of latitudes on one hemisphere
 
     # check that `nlat_half` match the vector `v` length
-    OctahedralGaussianGrid{T}(v,nlat_half) where T = length(v) == npoints_octahedral(nlat_half) ?
-    new(v,nlat_half) : error("$(length(v))-element Vector{$(eltype(v))}"*
+    OctahedralGaussianGrid{T}(data,nlat_half) where T = length(data) == npoints_octahedral(nlat_half,false) ?
+    new(data,nlat_half) : error("$(length(data))-element Vector{$(eltype(data))}"*
     "cannot be used to create a O$(nlat_half) OctahedralGaussianGrid{$T}.")
 end
 
 # number of points and longitudes per ring on the octahedral grid
-npoints_octahedral(nlat_half::Integer) = 4nlat_half^2 + 36nlat_half
-nlat_half_octahedral(npoints::Integer) = round(Int,-9/2+sqrt((9/2)^2 + npoints/4))  # inverse
+npoints_octahedral(nlat_half::Integer,nlat_oddp::Bool) =
+    nlat_oddp ? max(0,4nlat_half^2 + 32nlat_half - 16) : 4nlat_half^2 + 36nlat_half # max(0,...) needed to avoid negative array size when nlat_half==0
+nlat_half_octahedral(npoints::Integer,nlat_oddp::Bool) =
+    nlat_oddp ? round(Int,-4+sqrt(20 + npoints/4)) : round(Int,-9/2+sqrt((9/2)^2 + npoints/4))  # inverse
 nlon_octahedral(j::Integer) = 16+4j
+OctahedralGaussianGrid{T}(data::AbstractVector) where T = OctahedralGaussianGrid(data,nlat_half_octahedral(length(data),false))
+
+truncation_order(::Type{<:OctahedralGaussianGrid}) = 3      # cubic
+get_truncation(::Type{<:OctahedralGaussianGrid},nlat_half::Integer) = nlat_half-1
+get_resolution(::Type{<:OctahedralGaussianGrid},trunc::Integer) = roundup_fft(trunc+1)
+# get_nresolution() is already implemented for AbstractOctahedralGrid
+nlat_odd(::Type{<:OctahedralGaussianGrid}) = false
+# get_nlon() is already implemented for AbstractOctahedralGrid
+# get_nlon_per_ring() is already implemented for AbstractOctahedralGrid
+get_npoints(::Type{<:OctahedralGaussianGrid},nlat_half::Integer) = npoints_octahedral(nlat_half,false)
+get_colat(::Type{<:OctahedralGaussianGrid},nlat_half::Integer) = get_colat(FullGaussianGrid,nlat_half)
+# get_lon() is already implemented for AbstractOctaherdalGrid
+# get_colatlons() is already implemented for AbstractOctaherdalGrid
+# each_index_in_ring() is already implemented for AbstractOctaherdalGrid
+get_quadrature_weights(::Type{<:OctahedralGaussianGrid},nlat_half::Integer) = gaussian_weights(nlat_half)
```

Comment:
I've put a `get_solid_angles` function on top of `get_quadrature_weights`, such that in spectral_transform.jl we have one interface for all grids that give conceptually the same thing back. Before we had to scale some quadrature weights with nlon some not. Hence now, every grid also should provide  a `get_solid_angle` method which is in most cases `get_quadrature ./ nlon` but is directly 4pi/npoints for HEALPix. 

So there'll be some conflicts whichever branch is merged first, but we can resolve that

Reply:
Great! This is more elegant, again.

Reply:
merged in